tests/suites/test_suite_version.data

```diff
@@ -5,11 +5,13 @@ Check runtime library version
 check_runtime_version:"2.8.0"
 
 Check for MBEDTLS_VERSION_C
+depends_on:MBEDTLS_VERSION_FEATURES
```

Comment:
Since `check_feature` itself requires `MBEDTLS_VERSION_FEATURES`, the dependency should be in the `BEGIN_CASE` magic comment for `check_feature` in `test_suite_version.function`.

Reply:
well, i'm certainly not an expert, but this change seems to works as it is.
without it disabling `MBEDTLS_VERSION_C` breaks `test_suite_version`:

```
$ ./test_suite_version                                                                                                                                                                                        
Check compiletime library version ................................. PASS
Check runtime library version ..................................... PASS
Check for MBEDTLS_VERSION_C ....................................... FAILED
  check == result
  at line 71, suites/test_suite_version.function

Check for MBEDTLS_AES_C when already present ...................... FAILED
  check == result
  at line 71, suites/test_suite_version.function

Check for unknown define .......................................... FAILED
  check == result
  at line 71, suites/test_suite_version.function

----------------------------------------------------------------------------

FAILED (2 / 5 tests (0 skipped))
```

and applying it makes the test suite skip tests:

```
$ ./test_suite_version 
Check compiletime library version ................................. PASS
Check runtime library version ..................................... PASS
Check for MBEDTLS_VERSION_C ....................................... ----
Check for MBEDTLS_AES_C when already present ...................... ----
Check for unknown define .......................................... ----

----------------------------------------------------------------------------

PASSED (5 / 5 tests (3 skipped))
```

Reply:
It works this way, but it isn't how we want to maintain it: since the function requires `MBEDTLS_VERSION_FEATURES` (and might well become uncompilable without `MBEDTLS_VERSION_FEATURES`), the function itself should be excluded if `MBEDTLS_VERSION_FEATURES` is disabled.

Reply:
```mbedtls_version_check_feature``` is guarded by ```MBEDTLS_VERSION_C``` and not ```MBEDTLS_VERSION_FEATURES``` See https://github.com/ARMmbed/mbedtls/blob/development/library/version_features.c#L700

Reply:
@gilles-peskine-arm fair enough, done.

@mazimkhan that's true, but the actual list of features is guarded by `MBEDTLS_VERSION_FEATURES` ([here](https://github.com/ARMmbed/mbedtls/blob/f65379bc400707baac9466cce88a759685ac6c01/library/version_features.c#L35)) and the tests fail if the list is empty. this actually doesn't make much sense to me - what good does having `mbedtls_version_check_feature` make if the list of features is not populated? i'd put this function behind `MBEDTLS_VERSION_FEATURES` as well, that'd make more sense to me.

Reply:
Yes sure we can see it that way. But it doesn't break compilation.

Reply:
@mazimkhan um... sure it will but the 3 `check_feature` tests will fail if mbedTLS is compiled with `VERSION_C` but without `VERSION_FEATURES`, hence the dependency.

Reply:
The tests will not fail but skip if ```MBEDTLS_VERSION_FEATURES``` is a dependency in either ```.function``` or ```.data``` file. If it is omitted altogether then yes the tests will fail. The point here is where do we put the dependency. Since ```MBEDTLS_VERSION_FEATURES``` is functionally required in all those tests it is ok and less code to put it in the .function file.

Reply:
@mazimkhan ok... so what should i do? @gilles-peskine-arm asked me to put it into the `.function` file, so i did. do you want me to move it back to `.data`? or maybe put `MBEDTLS_VERSION_C` as a function dependency and then `MBEDTLS_VERSION_FEATURES` as particular testcase dependency in the .data file, is that what you'd like me to do?

Reply:
Please don't change anything. I am not saying Gilles' suggestion is wrong. I was trying to explain that your original code does not break compilation and Gilles' suggestion is also correct since ```mbedtls_version_check_feature``` functionally depends on ```MBEDTLS_VERSION_FEATURES```. 

Everything is fine. Sorry if I confused you.

Reply:
@mazimkhan not do anything - roger that, can do :)